server/models/Transaction.js

```diff
@@ -108,22 +135,26 @@ export default (Sequelize, DataTypes) => {
 
       createFromPayload({ transaction, user, group, paymentMethod }) {
 
-        // attach other objects manually. Needed for afterCreate hook to work properly
-        transaction.UserId = user && user.id;
-        transaction.GroupId = group && group.id;
-        transaction.PaymentMethodId = transaction.PaymentMethodId || (paymentMethod ? paymentMethod.id : null);
-        transaction.type = (transaction.amount > 0) ? type.DONATION : type.EXPENSE;
-
-        if (transaction.amount > 0 && transaction.txnCurrencyFxRate) {
-          // populate netAmountInGroupCurrency for donations
-            transaction.netAmountInGroupCurrency =
-              Math.round((transaction.amountInTxnCurrency
-                - transaction.platformFeeInTxnCurrency
-                - transaction.hostFeeInTxnCurrency
-                - transaction.paymentProcessorFeeInTxnCurrency)
-              *transaction.txnCurrencyFxRate);
-        }
-        return Transaction.create(transaction);
+        return group.getHost().then(host => {
+          transaction.HostId = host.id;
```

Comment:
Added this

    if (!host) {
      throw new Error(`Cannot create a transaction: collective id ${group.id} doesn't have a host`);
    }